pkg/globalnet/controllers/egress_pod_watcher.go

```
@@ -22,14 +22,27 @@ import (
 
 	"github.com/submariner-io/admiral/pkg/log"
 	"github.com/submariner-io/admiral/pkg/watcher"
+	"github.com/submariner-io/submariner/pkg/ipset"
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/cache"
 	"k8s.io/klog"
+	utilexec "k8s.io/utils/exec"
 )
 
-func startPodWatcher(name, namespace string, config watcher.Config) (*podWatcher, error) {
-	pw := &podWatcher{stopCh: make(chan struct{})}
+func startPodWatcher(name, namespace, ipsetName string, config watcher.Config, podSelector *metav1.LabelSelector) (*podWatcher, error) {
+	pw := &podWatcher{
+		stopCh:     make(chan struct{}),
+		ipSetName:  ipsetName,
+		ipSetIface: ipset.New(utilexec.New()),
+	}
+
+	var labelSelector string
+	if sel, err := metav1.LabelSelectorAsSelector(podSelector); err == nil {
```

We should report the error.